rafthttp/peer.go

```diff
@@ -100,6 +117,36 @@ func startPeer(tr http.RoundTripper, u string, local, to, cid types.ID, r Raft,
 				if err := r.Process(context.TODO(), mm); err != nil {
 					log.Printf("peer: process raft message error: %v", err)
 				}
+			case status := <-p.statusc:
+				name := status.name
```

Comment:
these all "work" related logic, should not be put into the peer code. should have a separate function to abstract out the logic.